File: fel-adapter/src/de/proskor/fel/impl/EventInstanceContainerImpl.scala
package de.proskor.fel.impl

import java.util.{List => JavaList}

import scala.collection.JavaConversions.seqAsJavaList

import de.proskor.automation.Element
import de.proskor.automation.Repository
import de.proskor.fel.container.EventInstanceContainer
import de.proskor.fel.container.EventTypeContainer
import de.proskor.fel.event.EventInstance

class EventInstanceContainerImpl(val peer: Element) extends EntityImpl(peer) with EventInstanceContainer {
  override def getType: EventTypeContainer = {
    val types = for {
      connector <- peer.connectors
      if connector.source == peer && connector.stereotype == "instanceOf"
      target = connector.target
    } yield new EventTypeContainerImpl(target)
    types.headOption.orNull
  }

  override def getEvents: JavaList[EventInstance] = {
    val events = for {
      element <- peer.elements if element.stereotype == "Event"
    } yield new EventInstanceImpl(element)
    events.toList
  }

  override def getChildren: JavaList[EventInstanceContainer] = List() // TODO
  override def getParent: EventInstanceContainer = null // TODO
}

File: ea-fel/src/de/proskor/fel/impl/stub/EventInstanceContainerImpl.java
package de.proskor.fel.impl.stub;

import java.util.ArrayList;
import java.util.List;

import de.proskor.fel.container.EventInstanceContainer;
import de.proskor.fel.container.EventTypeContainer;
import de.proskor.fel.event.EventInstance;

public class EventInstanceContainerImpl extends EntityImpl implements EventInstanceContainer {
	private EventTypeContainer type;
	private List<EventInstance> eventInstances = new ArrayList<EventInstance>();
	private List<EventInstanceContainer> children = new ArrayList<EventInstanceContainer>();
	private EventInstanceContainer parent;

	public EventTypeContainer getTyp() {
		return type;
	}

	public void setType(EventTypeContainer typ) {
		this.type = typ;
	}

	@Override
	public EventTypeContainer getType() {
		return this.type;
	}

	@Override
	public List<EventInstance> getEvents() {
		return this.eventInstances;
	}

	@Override
	public String getQualifiedName() {
		if (parent != null)
			return parent.getQualifiedName() + "." + getName();
		else
			return getName();
	}

	@Override
	public EventInstanceContainer getParent() {
		return parent;
	}

	@Override
	public List<EventInstanceContainer> getChildren() {
		return children;
	}
}
